tests/cli/dorq/test_arg_resolvers.py

```diff
@@ -653,6 +653,276 @@ def return_wf(id, time_delay_in_sec: int):
             # Resolver should return the user's choice.
             assert resolved_run == selected_run
 
+    class TestResolveLogSwitches:
+        @staticmethod
+        @pytest.mark.parametrize(
+            "switches",
+            [
+                (False, False, True),
+                (False, True, False),
+                (False, True, True),
+                (True, False, False),
+                (True, False, True),
+                (True, True, False),
+                (True, True, True),
+            ],
+        )
+        def test_returns_unchanged_if_all_switches_are_set_and_all_logs_available(
+            switches,
+        ):
+            """
+            The most trivial case - all of the log types are available, and the user
+            has set every switch. Under these circumstances the resolver should do
+            nothing.
+            """
+            # Given
+            logs = Mock(per_task=["foo"], system=["bar"], env_setup=["baz"])
```

Comment:
**question**: is there a reason to use `Mock` here? [WorkflowLogs](https://github.com/zapatacomputing/orquestra-workflow-sdk/blob/7161937c7d782d7e6152b01b04da452c5fa7fe3b/src/orquestra/sdk/_base/_logs/_interfaces.py#L15) is a simple record type.